pkg/pidfile/pidfile.go

```diff
@@ -0,0 +1,75 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Tetragon
+package pidfile
+
+import (
+	"bytes"
+	"errors"
+	"os"
+	"path/filepath"
+	"strconv"
+
+	"github.com/cilium/tetragon/pkg/defaults"
+	"github.com/cilium/tetragon/pkg/option"
+	"github.com/cilium/tetragon/pkg/reader/proc"
+)
+
+var (
+	ErrPidFileNotPresent = errors.New("pid file is not present")
+	ErrPidIsNotAlive     = errors.New("process is not alive")
+	ErrPidIsStillAlive   = errors.New("process is already running")
+)
+
+func readPidFile() (uint64, error) {
+	_, err := os.Stat(defaults.DefaultPidFile)
+	if err != nil {
+		return 0, ErrPidFileNotPresent
+	}
+
+	data, err := os.ReadFile(defaults.DefaultPidFile)
```

Comment:
you can check for returned error from `os.ReadFile` (https://pkg.go.dev/os#pkg-variables) and it can return: ErrNotExist   = [fs](https://pkg.go.dev/io/fs).[ErrNotExist](https://pkg.go.dev/io/fs#ErrNotExist). You can remove the `os.Stat` then.

Reply:
Alright done ;-)